core/templates/pages/diagnostic-test-player-page/diagnostic-test-current-topic-status.model.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview The model keeps track of the current topic that is being
+ * tested in any instance of the diagnostic test. This model maintains the
+ * skill IDs associated with which the next question should be presented
+ * in the diagnostic test.
+ */
+
+import { Question } from 'domain/question/QuestionObjectFactory';
+
+export interface SkillIdToQuestionsDict {
+  [skillId: string]: {
+    mainQuestion: Question;
+    backupQuestion: Question;
+  };
+}
+
+export class DiagnosticTestCurrentTopicStatusModel {
+  // A dict with skill ID as key and a nested dict as value. The nested dict
+  // contains the main question and back question as two keys and the Question
+  // object as the values of each. The main question is the first question
+  // that the learner encounters in the diagnostic test and if they attempted
+  // the main question incorrectly the backup question will be presented
```

Comment:
...will be presented. Otherwise, the backup ...

Reply:
Done